hil/ext/switches/_dell_base.py

```diff
@@ -36,37 +37,74 @@ def disable_native(self, vlan_id):
         self._sendline('sw trunk native vlan none')
 
     def get_port_networks(self, ports):
-        num_re = re.compile(r'(\d+)')
+        '''Returns every trunking VLAN and native VLAN from a port config.
+        Example: Port 1 has 'Trunking Native Mode VLAN': ' 3 (Inactive)',
+        'Trunking VLANs Enabled': 'none'
+        Port 2 has 'Trunking Native Mode VLAN': 'none',
+        'Trunking VLANs Enabled': ' 10, 2000'
+        Port 3 has 'Trunking Native Mode VLAN': 'none',
+        'Trunking VLANs Enabled': ' 23 - 25'
+
+        Return:
+        {<hil.model.Port object at 0xDEADBEEF0123>: [('vlan/native', '3')],
+        <hil.model.Port object at 0x7F8503AB4991>: [('vlan/10', 10),
+                                                    ('vlan/2000', 2000)],
+        <hil.model.Port object at 0xABCD01234ABCD>: [('vlan/23', 23),
+                                                     ('vlan/24', 24),
+                                                     ('vlan/25', 25)]}
+        '''
         port_configs = self._port_configs(ports)
+        # iterate through the port configurations
         result = {}
         for k, v in port_configs.iteritems():
-            native = v['Trunking Native Mode VLAN'].strip()
-            match = re.match(num_re, native)
-            if match:
-                # We need to call groups to get the part of the string that
-                # actually matched, because it could include some junk on the
-                # end, e.g. "100 (Inactive)".
-                num_str = match.groups()[0]
-                native = int(num_str)
+            network_list = []
+            non_native_list = []
+            # get native vlan then remove junk if native is not None
+            try:
+                native_vlan = v['Trunking Native Mode VLAN'].strip()
+            except KeyError:
+                # some dell switches say "Mode Native" instead
+                native_vlan = v['Trunking Mode Native VLAN'].strip()
+            # append native vlan if appropriate:
+            if native_vlan != 'none':
+                native_vlan = native_vlan.replace(' (Inactive)', '')
+                if hasattr(self.switch, 'dummy_vlan'):
+                    # ignore if dummy vlan
+                    if int(native_vlan) == int(self.switch.dummy_vlan):
+                        native_vlan = None
+                    else:
+                        network_list.append(('vlan/native', int(native_vlan)))
+                else:
+                    # appropriate to append native vlan to list
+                    network_list.append(('vlan/native', int(native_vlan)))
```

Comment:
we could probably collapse the 2 if statements into 1:
```python
if hasattr(self.switch, 'dummy_vlan') and int(native_vlan) == int(self.switch.dummy_vlan):
    native_vlan = None
else:
    network_list.append(('vlan/native', int(native_vlan)))
````

that way we don't have 2 repeated else blocks either.